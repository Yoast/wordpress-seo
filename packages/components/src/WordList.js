// External dependencies.
import React from "react";
import PropTypes from "prop-types";

/**
 * @summary WordList component.
 *
 * @deprecated
 *
 * @param {object}      props                 The props to use.
 * @param {string}      props.title           The title of the list.
 * @param {string}      props.classNamePrefix CSS classname prefix for the elements.
 * @param {array}       props.words           The relevant words.
 * @param {HTMLElement} props.header          Function to call before list is shown.
 * @param {HTMLElement} props.footer          Function to call after list is shown.
 *
 * @returns {JSX.Element} Rendered WordList component.
 */
const WordList = ( props ) => {
	console.warn( "The WordList component has been deprecated and will be removed in a future release." );

	const { title, classNamePrefix, words, header, footer } = props;

	const list = (
		<ol className={ classNamePrefix + "__list" }>
			{ words.map( word => {
				return (
					<li
						key={ word }
						className={ classNamePrefix + "__item" }
					>
						{ word }
					</li>
				);
			} ) }
		</ol>
	);

	return (
		<div className={ classNamePrefix }>
			<p><strong>{ title }</strong></p>
			{ header }
			{ list }
			{ footer }
		</div>
	);
};

WordList.propTypes = {
	words: PropTypes.array.isRequired,
	title: PropTypes.string.isRequired,
	header: PropTypes.string,
	footer: PropTypes.string,
	classNamePrefix: PropTypes.string,
};

WordList.defaultProps = {
	classNamePrefix: "",
	header: "",
	footer: "",
};

export default WordList;
