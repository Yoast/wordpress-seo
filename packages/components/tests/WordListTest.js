import React from "react";
import renderer from "react-test-renderer";

import WordList from "../src/WordList";

describe( "WordList", function() {
	it( "renders wordlist as list items", () => {
		const words = [ "word1", "word2", "word3" ];

		const wordlist = renderer.create(
			<WordList title="Wordlist" words={ words } />
		);

		const tree = wordlist.toJSON();
		expect( tree ).toMatchSnapshot();
	} );

	it( "renders correctly without items", () => {
		const words = [];

		const wordlist = renderer.create(
			<WordList title="Wordlist" words={ words } />
		);

		const tree = wordlist.toJSON();
		expect( tree ).toMatchSnapshot();
	} );
} );
