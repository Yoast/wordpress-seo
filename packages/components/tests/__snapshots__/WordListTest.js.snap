// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WordList renders correctly without items 1`] = `
<div
  className=""
>
  <p>
    <strong>
      Wordlist
    </strong>
  </p>
  
  <ol
    className="__list"
  />
  
</div>
`;

exports[`WordList renders wordlist as list items 1`] = `
<div
  className=""
>
  <p>
    <strong>
      Wordlist
    </strong>
  </p>
  
  <ol
    className="__list"
  >
    <li
      className="__item"
    >
      word1
    </li>
    <li
      className="__item"
    >
      word2
    </li>
    <li
      className="__item"
    >
      word3
    </li>
  </ol>
  
</div>
`;
